src/backend/vm/runtime.rs

```diff
@@ -337,5 +404,113 @@ mod tests {
         } else {
             panic!("Wrong type");
         }
+        if let Data::Int(data) = vm.load_register(7).unwrap() {
```

Comment:
Can we make this a helper macro?
```
macro_rules! assert_reg {
    ($reg:expr, $variant: ident, $assert:expr, $vm:ident ) => {
        if let Data::$variant(data) = $vm.load_register(7).unwrap() {
            assert_eq!(data, $assert);
        }else{
            panic!("Wrong type!");
        }
    };
}
```
which can be used like 
```suggestion
                assert_reg!(7, Int, 23/12, vm);
```

Reply:
I think it would be better to abstract out "data equivalence tests". Please see @permui [abstract data equivalence tests](https://github.com/diatom-lang/diatom/pull/3/commits/06dca8f36218d67e991b3a2ff5331245ee50f6bd).